### DISCUSSION TITLE
How to disable aerospace key binds when a given application is in focus?

### DISCUSSION THREAD

**Discussion by ChrisGVE** [nikitabobko/AeroSpace#769] [questions-and-answers]
Hello
My situation is as follows: when working on my Mac, I use Aerospace, like you all, to organize my screen. However, there are times when I work from home that I use Citrix to connect to my office's virtual Windows machine, which has its own keymaps, and they interfere with Aerospace (on purpose, I must say, so that I can use the same with Aerospace).

My problem is that for the Citrix connection to work as intended, I must quit Aerospace while I am connected to work and return to Aerospace when I am done. This causes all sorts of problems (my Aerospace config is not finalized, and I am only starting to get familiar with it).

So, in short, is there an out-of-the-box solution to mask Aerospace users' configured key bindings by application?

**Comment by Dimentium** [comment]
> Hello
> 
> My situation is as follows: when working on my Mac, I use Aerospace, like you all, to organize my screen. However, there are times when I work from home that I use Citrix to connect to my office's virtual Windows machine, which has its own keymaps, and they interfere with Aerospace (on purpose, I must say, so that I can use the same with Aerospace).
> 
> 
> 
> My problem is that for the Citrix connection to work as intended, I must quit Aerospace while I am connected to work and return to Aerospace when I am done. This causes all sorts of problems (my Aerospace config is not finalized, and I am only starting to get familiar with it).
> 
> 
> 
> So, in short, is there an out-of-the-box solution to mask Aerospace users' configured key bindings by application?

Hi,

I believe that this is again a problem similar to the ones raised in discussions #640 and #767, and my proposed solution is to use a separate keyboard shortcut manager (such as Karabiner, skhd, Hammerspoon) rather than expecting this functionality in the window manager.

By using Karabiner, you can flexibly configure the operation of keyboard shortcuts based on conditions such as the keyboard, active application, active layout, and pressed modifier.

**Reply by ChrisGVE** [reply]
I see. That's a way to look at it, but it just increases complexity and creates problems on its own. I would much rather prefer a solution similar to Kitty, which can condition its keymaps, such as when using the smart-split plugin. Having this feature in Aerospace (if one does not already exist) would be really handy.

**Reply by ChrisGVE** [reply]
Though looking at your example in [#640](https://github.com/nikitabobko/AeroSpace/discussions/640), I see a lot of potential for great stuff; unfortunately, it does increase the complexity of maintaining a configuration, and in my case, it is a bit like using a cannon to kill a mosquito 😉
